### PR TITLE
Extend a common base type for various related driver error types

### PR DESCRIPTION
**Pull request by markfields**

### CODE REVIEW

File: packages/drivers/odsp-driver-definitions/src/errors.ts [hunk 1]
```diff
@@ -52,11 +52,12 @@ export enum OdspErrorType {
  * Base interface for all errors and warnings
  * Superset of IDriverErrorBase, but with Odsp-specific errorType
  */
-export interface IOdspError {
+export interface IOdspError extends Omit<IDriverErrorBase, "errorType"> {
```
Comment (anthony-murphy) [inline packages/drivers/odsp-driver-definitions/src/errors.ts:55]: why not IAnyDriverError  here?

Comment (markfields) [reply packages/drivers/odsp-driver-definitions/src/errors.ts:55]: Probably dependencies.  `IAnyDriverError` is in `driver-utils` at the moment, mostly for pragmatic reasons.  Would you agree with moving it to driver-definitions?

Comment (anthony-murphy) [reply packages/drivers/odsp-driver-definitions/src/errors.ts:55]: i'd rather not add anything to definitions with the current approach, as that is just more to deprecate when we move to error api

**Review by anthony-murphy — APPROVED** [review]

File: packages/drivers/odsp-driver-definitions/src/errors.ts (hunk 1 above)
Comment (markfields) [reply packages/drivers/odsp-driver-definitions/src/errors.ts:55]: Sounds good to me

**Pull request merged**